### ISSUE TITLE
CS8509 (Switch is not exhaustive): Make it smarter.

### ISSUE THREAD

**Issue by TonyValenti** [dotnet/roslyn#70164]
**Brief description:**

CS8509 can trigger false positives when all possible cases are handled.

```csharp
var x = ...
if(x is 1 or 2){ 
 //This should not trigger CS8509 because only the handled cases can get to this branch
  var value = x switch {
   1 => true,
   2 => false,
  };

}
```

**Code example that the analyzer should report:**
It should NOT report the above.

**Languages applicable:**
C#

**Additional information:**
N/A

**Documentation requirements:**
N/A

**Comment by CyrusNajmabadi** [comment]
This would be a request for the language.  Moving there.
